### PR TITLE
CODEOWNERS: add file, MikeMcQuaid.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
I don't watch Homebrew/homebrew-core any more (as other maintainers are doing a great job of the formula work without my help) but I do want to ensure that I keep on top of changes to certain files.

As a result, add a `CODEOWNERS` file so I can be requested for review on changes to these files: https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

@Homebrew/core any objections to the use of this pattern or the current file contents? Similarly: if any of you would like to be requested for review on any files or formulae (you/I won't "own" them regardless): feel free to suggest here or submit PRs to this file after it is merged. Another option we could consider would be using GitHub teams for any/all of these instead (e.g. I could see @Homebrew/plc wanting to be consulted on any license file changes).

### CODE REVIEW

**Review by reitermarkus — APPROVED** [review]

**Comment by issyl0** [comment]
Are we going to set the "require review by CODEOWNERS" setting or is this just "I want to know that a thing has changed"?

**Comment by MikeMcQuaid** [comment]
> is this just "I want to know that a thing has changed"?

Yeh, this. I don't think it makes sense to block on a single maintainer/team but I think it would be good to have some maintainers/teams notified on review requests. This can also be used to filter specific emails for "high-priority" files/formulae for given maintainers.

**Comment by jonchang** [comment]
In my capacity as a PLC member: I think the PLC should definitely be a code owner on important documents like LICENSE. This should be extended to relevant stuff in Homebrew/brew as well.

In my capacity as a maintainer: I think if maintainers want to add themselves as CODEOWNERS of particularly tricky formulae e.g., `rust`, `go`, `meson`, `gtk*`, `openjdk*`, and so on, that's up to them. But I think we need to clarify in our contributing documents that when applied to formulae, CODEOWNER doesn’t literally mean that that specific maintainer must sign off on a review, it's more of an advisory thing.

**Comment by MikeMcQuaid** [comment]
> Homebrew/core and homebrew/cask teams should definitely be CODEOWNERS on tap_migrations.json to ensure we aren't migrating stuff to the wrong cask / out of the Homebrew org.

Does this come up often enough to warrant it?

**Comment by MikeMcQuaid** [comment]
I've added comments to this file to clarify intent and who should/shouldn't be added in future.

**Review by SMillerDev — APPROVED** [review]

File: CODEOWNERS
Comment (issyl0) [inline CODEOWNERS:2]:
```suggestion
# people with write-access to this repository (i.e. @Homebrew/core) who wish to be notified of changes to these
```

**Comment by issyl0** [comment]
You'll also get notifications for PRs to change these files in Homebrew/linuxbrew-core, not just Homebrew/homebrew-core, because of the way the merge process works. Not sure how many there are, as most of these things are defined upstream (here) anyway, but just making you aware...

**Comment by SMillerDev** [comment]
I think linuxbrew changing the license independently might be good to be notified of 👍

**Review by chenrui333 — APPROVED** [review]

**Review by sjackman — APPROVED** [review]

File: CODEOWNERS
Comment (samford) [inline CODEOWNERS:1]:
```suggestion
# Note that the naming of this file is incorrect for our case: these people do not "own" the provided files but are
```

**Comment by MikeMcQuaid** [comment]
Thanks for review, all!

**Review by canniffamber — COMMENTED** [review]
Tu

**Pull request merged**
